src/toil/common.py

```diff
@@ -1268,6 +1273,36 @@ def getToilWorkDir(configWorkDir: Optional[str] = None) -> str:
             raise RuntimeError(f'The directory specified by --workDir or TOIL_WORKDIR ({workDir}) does not exist.')
         return workDir
 
+    @staticmethod
+    def get_working_tmpdir(tmpdir_prefix: Optional[str] = None) -> Optional[str]:
+        """
+        Get a working temp directory, testing if it is accessible
+        If tmpdir_prefix is given, assuming it is accessible, it will return its parent directory
+        if the path itself is not a directory path
+        :param tmpdir_prefix: tmp directory to override
+        :return: Path or none
+        """
+        # if tmpdir_prefix exists, return its parent or None depending on if it is accessible
+        if tmpdir_prefix is not None:
+            return tmpdir_prefix if os.path.isdir(tmpdir_prefix) else os.path.dirname(tmpdir_prefix)
```

Comment:
I don't think we want `tmpdir_prefix if os.path.isdir(tmpdir_prefix)`. If `tmpdir_prefix` is `/tmp/cwltmp-` but the directory `/tmp/cwltmp-` happens to exist, we still want to make temproary files and directories like `/tmp/cwltmp-XXXX` and not change to making them *in* that directory. The CWL-style tmpdir prefix will end in `/` if it is itself a directory we are meant to make stuff in.